### PR TITLE
Fix: Allow create_external_models to run for projects with star projections

### PR DESCRIPTION
**Pull request by crericha**
Also support external models in dbt formatted projects.

### CODE REVIEW

File: sqlmesh/dbt/loader.py [hunk 1]
```diff
@@ -102,6 +102,8 @@ def _load_models(
                 }
             )
 
+        models.update(self._load_external_models())
```
Comment (izeigerman) [inline sqlmesh/dbt/loader.py:105]: Shouldn't `_load_external_models` be called in the `load` directly and then we wouldn't need to call it explicitly here?

File: sqlmesh/core/loader.py [hunk 2]
```diff
@@ -97,7 +99,7 @@ def __init__(self) -> None:
         self._path_mtimes: t.Dict[Path, float] = {}
         self._dag: DAG[str] = DAG()
 
-    def load(self, context: Context) -> LoadedProject:
+    def load(self, context: Context, update_schemas: bool = True) -> LoadedProject:
```
Comment (izeigerman) [inline sqlmesh/core/loader.py:102]: [Nit] Missing docstring for the new argument.

File: sqlmesh/core/loader.py [hunk 3]
```diff
@@ -69,7 +69,9 @@ def update_model_schemas(dag: DAG[str], models: UniqueKeyDict[str, Model]) -> No
         if external:
             if "*" in model.columns_to_types:
                 raise ConfigError(
-                    f"Can't expand SELECT * expression for model '{name}'. Projections for models that use external sources must be specified explicitly at '{model._path}'"
+                    f"Can't expand SELECT * expression for model '{name}'."
+                    " Projections for models that use external sources must be specified explicitly"
```
Comment (izeigerman) [inline sqlmesh/core/loader.py:73]: Perhaps include which command to run to fix the issue.

File: sqlmesh/core/context.py [hunk 4]
```diff
@@ -848,12 +848,13 @@ def create_external_models(self) -> None:
         The schema file contains all columns and types of external models, allowing for more robust
         lineage, validation, and optimizations.
         """
+        models = self._models or self._loader.load(self, update_schemas=False).models
```
Comment (izeigerman) [inline sqlmesh/core/context.py:851]: So it seems like in this case we only need to load models but we'd load everything instead.  

If we made `_load_models` public and kept everything else unchanged we'd only load models and we wouldn't require the `update_schemas` flag.

Comment (crericha) [reply sqlmesh/core/context.py:851]: I looked at that first too. _load_models requires macros, etc., So it felt unnecessarily complicated to load each bit manually to save time on a one-off command.

File: sqlmesh/core/loader.py (hunk 3 above)
Comment (crericha) [reply sqlmesh/core/loader.py:73]: I link to the external models docs page, which includes the CLI command. Do you think more is needed?

File: sqlmesh/dbt/loader.py (hunk 1 above)
Comment (crericha) [reply sqlmesh/dbt/loader.py:105]: Yeah, I can do it that way. It'd still look like `models.update(super()._load_models(args))`, so I thought the code was simpler by duplicating the one line, but I can use inheritance if you prefer. Let me know.

File: sqlmesh/core/loader.py (hunk 3 above)
Comment (izeigerman) [reply sqlmesh/core/loader.py:73]: Yeah, I think they should be able to try something immediately without having to read the lengthy doc page. If the quick way doesn't work for them, then they can check out the doc page.

**Pull request merged**